go/state/ffistate/ffi_state.go

```diff
@@ -72,122 +252,137 @@ func newState(impl C.enum_LiveImpl, params state.Parameters) (state.State, error
 		return nil, fmt.Errorf("%w: unsupported archive type %v", state.UnsupportedConfiguration, params.Archive)
 	}
 
+	var ffi FFIInterface
+
+	switch ffiType {
+	case FFITypeCpp:
+		ffi = CppInterface{}
+	case FFITypeRust:
+		ffi = RustInterface{}
+	default:
+		panic("unsupported FFI type")
```

Comment:
No sure what the policy on panicking in Go is. If we hit this case, we definitely have a bug and can not reasonably recover. The question is, if we are supposed to panic here or return an error and panic further up

Reply:
Done!